contracts/StrategyCurve3crv.sol

```diff
@@ -288,27 +341,62 @@ contract StrategyCurve3crv is StrategyCurveBase {
         }
 
         // we're done harvesting, so reset our trigger if we used it
-        forceHarvestTriggerOnce = false;
+        delete forceHarvestTriggerOnce;
+    }
+
+    /* ========== ySwaps ========== */
+
+    /** 
+    @notice Update Trade Handler (ySwaps) address, approve handler contract to sell our 
+    reward tokens, and register the swaps required by this strategy.
+    @param _tradeHandler ySwaps contract address
+    */
+    function updateTradeHandler(address _tradeHandler) external onlyGovernance {
+        _removeTradeHandlerPermissions();
+        tradeHandler = _tradeHandler;
+        _enableTrades();
+        emit UpdatedTradeHandler();
+    }
+
+    /** 
+    @notice Takes care of the approvals needed by Trade Handler (ySwaps) and 
+    enables the swaps required by the strategy
+    */
+    function _enableTrades() internal {
+        // approve and set up trade handler
+        crv.safeApprove(tradeHandler, type(uint256).max);
+        gno.safeApprove(tradeHandler, type(uint256).max);
+
+        ITradeHandler(tradeHandler).enable(address(crv), address(want));
+        ITradeHandler(tradeHandler).enable(address(gno), address(want));
     }
 
-    // Sells our CRV or GNO for our target token
-    function _sellToken(address token, uint256 _amount) internal {
-        if (token != address(targetToken)) {
-            address[] memory tokenPath = new address[](2);
-            tokenPath[0] = address(token);
-            tokenPath[1] = address(targetToken);
-            router.swapExactTokensForTokens(
-                _amount,
-                uint256(0),
-                tokenPath,
-                address(this),
-                block.timestamp
-            );
+    /** 
+    @notice Revoke Trade Handler (ySwaps) contract. Sets allowance to 0 and `tradeHandler`
+    to address(0)
+    */
+    function removeTradeHandlerPermissions() external onlyEmergencyAuthorized {
```

Comment:
I would call this method `disableTradeHandler()` since it will only be used in an emergency situation.